splash/browser_tab.py

```diff
@@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = str(uuid.uuid1())
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class HTMLElement(object):
+    DIMENSIONS_JS_FUNC = """
+        (function(element) {
+            var rect = element.getBoundingClientRect();
```

Comment:
Consider this link: ![screenshoot](http://i.imgur.com/hgVDv9x.png)

The blue semi-transparent rectangles are the client rects (it has two), the dotted line the bounding client rect (the rect that contains the client rects).

The top left corner of the bounding client rect is not "on" the element, the top-left of one of the client rects is.